src/Data/Matrix.purs

```diff
@@ -82,48 +124,77 @@ instance applyMat :: Apply (Mat s) where
 -- >     ( 0 0 ... 1 0 )
 -- >   n ( 0 0 ... 0 1 )
 --
-identity' :: forall s.  (Sized s) => Mat s Number
+identity' :: forall r c. Sized r => Sized c => Mat r c Number
 identity' = generate \ i j -> if i == j then 1.0 else 0.0
```

Comment:
I would generalize this to any `Semiring` (which provides `one` and `zero`, including for `Int` and `Number`):
```
identity' :: forall a r c. Semiring a => Sized r => Sized c => Mat r c a
  identity' = generate \i j -> if i == j then one else zero
```